core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -141,6 +143,41 @@ public static int getPrimaryColumnIndex(int base, int index) {
         return 2 + base + index * 2;
     }
 
+    /**
+     * Estimate if it's possible to do open all the files using this reader.
+     * Check that there is still enough of Open File Limits and Map limits left to open all partitions.
+     * If capacity is estimated to be not enough, CairoException will the thrown.
+     */
+    public void checkOsLimitsCapacity() {
```

Comment:
can we maintain count of non-zero partitions so that we don't have to iterate this list?

Reply:
done in https://github.com/questdb/questdb/pull/2289/commits/945b687d19f7d913e2abb6ccb33a663638ecdb5c